test/e2e/instrumentation/events.go

```diff
@@ -0,0 +1,191 @@
+
+	"github.com/onsi/ginkgo"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	eventRetryPeriod  = 1 * time.Second
+	eventRetryTimeout = 1 * time.Minute
+)
+
+var _ = ginkgo.Describe("[sig-instrumentation] Events", func() {
+	f := framework.NewDefaultFramework("events")
+
+	/*
+			   Release : v1.19
+			   Testname: Event resource lifecycle
+			   Description: Create an event, the event MUST exist.
+		           The event is patched with a new message, the check MUST have the update message.
+		           The event is deleted and MUST NOT show up when listing all events.
+	*/
+	ginkgo.It("should ensure that an event can be fetched, patched, deleted, and listed", func() {
+		eventTestName := "event-test"
+		someTime := metav1.MicroTime{Time: time.Unix(1505828956, 0)}
+
+		ginkgo.By("creating a test event")
+		// create a test event in test namespace
+		_, err := f.ClientSet.EventsV1beta1().Events(f.Namespace.Name).Create(context.TODO(), &v1beta1.Event{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: eventTestName,
+				Labels: map[string]string{
+					"testevent-constant": "true",
+				},
+			},
+			EventTime:           someTime,
+			Note:                "This is a test event",
+			Reason:              "Test",
+			Type:                "Normal",
+			ReportingController: "test-controller",
+			ReportingInstance:   "test-node",
+		}, metav1.CreateOptions{})
+		framework.ExpectNoError(err, "failed to create test event")
+
+		ginkgo.By("listing all events in all namespaces")
+		// get a list of Events in all namespaces to ensure endpoint coverage
+		eventsList, err := f.ClientSet.EventsV1beta1().Events("").List(context.TODO(), metav1.ListOptions{
+			LabelSelector: "testevent-constant=true",
+		})
+		framework.ExpectNoError(err, "failed list all events")
+
+		foundCreatedEvent := false
+		for _, val := range eventsList.Items {
+			if val.ObjectMeta.Name == eventTestName && val.ObjectMeta.Namespace == f.Namespace.Name {
+				foundCreatedEvent = true
+				break
+			}
+		}
+		framework.ExpectEqual(foundCreatedEvent, true, "unable to find the test event")
+
+		ginkgo.By("patching the test event")
+		// patch the event's note
+		eventPatchNote := "This is a test event - patched"
+		eventPatch, err := json.Marshal(map[string]interface{}{
+			"note": eventPatchNote,
+		})
+		framework.ExpectNoError(err, "failed to marshal the patch JSON payload")
+
+		_, err = f.ClientSet.EventsV1beta1().Events(f.Namespace.Name).Patch(context.TODO(), eventTestName, types.StrategicMergePatchType, []byte(eventPatch), metav1.PatchOptions{})
+		framework.ExpectNoError(err, "failed to patch the test event")
+
+		ginkgo.By("getting the test event")
+		// get event by name
+		event, err := f.ClientSet.EventsV1beta1().Events(f.Namespace.Name).Get(context.TODO(), eventTestName, metav1.GetOptions{})
+		framework.ExpectNoError(err, "failed to fetch the test event")
+		framework.ExpectEqual(event.Note, eventPatchNote, "test event note does not match patch note")
```

Comment:
We should probably compare the whole event to ensure that other stuff wasn't overwritten?

Reply:
Done.